### PR TITLE
README improvements

### PR DESCRIPTION
**Pull request by starkbamse**
**Changes**

- Improved visual style of documentation
- Added table of contents
- Added "Kaspa On Rust" logo
- Simplified the steps to setup a node
- Removed redundant description of utilizing nodejs WASM. 

**Why**

tldr; I improved the readability of the readme.

Simplifying the documentation and making it more accessible is key for Kaspa adoption. When I first came to this repo a couple days ago, I was struck that such a cool network had such an unorganized repository. I believe that the proposed changes will simplify the process for new developers to get started. 

These changes are obviously not perfect and I think there could be more information added regarding the WASM so that front-end developers that have little to no experience with complex languages like Rust to get engaged as well. :)

### CODE REVIEW

**Comment by starkbamse** [comment]
@aspect

**Comment by CaffeinatedFunctionality** [comment]
Love what you did here. Readmes have been super undervalued lately and aren't given a lot of love due to other developments.

**Comment by starkbamse** [comment]
> Love what you did here. Readmes have been super undervalued lately and aren't given a lot of love due to other developments.

Thanks, I totally agree and feel like README's are one of the most important parts in a repo since they are the first impression of the project.

**Comment by aspect** [comment]
Apologies for delayed feedback.

This looks good, but I would like to request the following changes.

1) for now, let's remove the image from the top of the document.  there is no consensus and the feedback for that specific image wasn't great.  so we will think about it, but for now this should go.

2) Please refactor linux instructions to be presented as multi-line code.  I think this can be done by adding back-slashes like so:

```
apt-get install clang-format clang-tidy \
clang-tools clang clangd libc++-dev \
libc++1 libc++abi-dev libc++abi1 \
...
```

The fact that this code now offers copy button is great, but the fact that you can not see the whole set of instructions on the screen and have to scroll for it is not good.  So we need to make this fit on the screen.

3) I wanted to do this for a while, perhaps you can restructure the document to separate each platform section into `<details>` and `<summary>` so that each section becomes an openable section.  

For example:
```
<details>
<summary>Installation on MacOS</summary>
... macos instructions
</details>
```
Then the other sections could perhaps be grouped this way as well.  IMO this will significantly reduce the footprint of the README making it much cleaner.  

For example, there can be a set of details sections for OS-specific installation + WASM + wRPC, then "Running the node" and "Mining" as well as "Simpa" can be in the open.

4) there need to be changes to include `sudo` in all appropriate linux-related commands as per https://github.com/kaspanet/rusty-kaspa/pull/341

**Comment by starkbamse** [comment]
@aspect 

Good points! Refactored in accordance with your suggestions. Includes the edits made by @coderofstuff in #341, added as co-author. Therefore also closes #340

**Comment by aspect** [comment]
So now this looks the opposite side of the spectrum.  There needs to be a balance of useful material presented and technical material being hidden.  Here is the list of sections (in a specific order) that should be visible or hidden:

1) Remove TOC, there is no point of having TOC to a bunch of hidden sections.  Hidden sections are a form of TOC.  I think perhaps the TOC is what forced you to create headings to be able to link to this.  If the document is structured properly with hidden sections, there is no need for TOC.

- Installation
  - Building on Linux (hidden)
  - Building on Windows (hidden)
  - Building on MacOS (hidden)
  - Building WASM framework (hidden)
  - Wallet CLI (hidden)
- Running the node
  - Start mainnet (open)
  - Start testnet (open)
  - wRPC
  - Mining
- Benchmarking & Testing
  - Simulation Framework (Simpa)
  - Heap Profiling (closed)
  - Tests (closed)
    - Run unit and most integration tests 
    - Using nextest
  - Benchmarks (closed)
  - Logging

So that is roughly 3 top-level sections (3 H1 headings), where Installation and Benchmarking is closed (thus it looks like TOC), and the main focus is on Running the node, where instructions are open by default.

**Comment by starkbamse** [comment]
@aspect The documentation has been adjusted I have grouped everything and hidden everything except for how to run the node. I added an intro text that I wrote with the help of an LLM. Let me know your thoughts on it. I believe it can serve as a starting point.

File: kaspa.png
Comment (coderofstuff) [inline kaspa.png:1]: Please remove the image from the PR

File: README.md [hunk 1]
```diff
@@ -1,235 +1,436 @@
...
-- Install wasm32 target: `rustup target add wasm32-unknown-unknown`
-- Run the following commands:
 
-```bash
-$ git clone https://github.com/kaspanet/rusty-kaspa
-$ cd rusty-kaspa
-```
+<h1 align="center">Kaspa On Rust</h1>
+
+Welcome to the Rust-based implementation of the Kaspa full-node and its ancillary libraries. This Alpha release serves as a replacement to the established <a href="https://github.com/kaspanet/kaspad">Golang node</a> (once the rust rewrite is completed), introducing developers to the possibilities of Rust in the Kaspa network's context.
```
Comment (coderofstuff) [inline README.md:4]: ...serves as a *drop-in* replacement...

File: README.md [hunk 2]
```diff
@@ -1,235 +1,436 @@
...
-- If you already have rust installed, update it by running: `rustup update`
-- Install wasm-pack: `cargo install wasm-pack`
-- Install wasm32 target: `rustup target add wasm32-unknown-unknown`
-- Run the following commands:
 
-```bash
-$ git clone https://github.com/kaspanet/rusty-kaspa
-$ cd rusty-kaspa
-```
+<h1 align="center">Kaspa On Rust</h1>
```
Comment (coderofstuff) [inline README.md:2]: This is the only piece of text that's aligned center. I think we should align it left to make the look overall consistent.

File: README.md [hunk 3]
```diff
@@ -1,235 +1,436 @@
...
+
+Welcome to the Rust-based implementation of the Kaspa full-node and its ancillary libraries. This Alpha release serves as a replacement to the established <a href="https://github.com/kaspanet/kaspad">Golang node</a> (once the rust rewrite is completed), introducing developers to the possibilities of Rust in the Kaspa network's context.
+
+Our codebase is an experimental ground for developers aiming to push the boundaries of blockchain scalability and speed. The Kaspa node aims to validate the feasibility of a high-speed blockchain, boasting block intervals of one second with prospects for significant enhancements.
```
Comment (coderofstuff) [inline README.md:6]: I think paragraph 2 (Our codebase...) and paragraph 3 (Contributors to this repo...) currently don't add too much value and should be removed. Ideally, the introduction section is small enough so we're not immediately seeing a wall of text.

In a different PR, after this is merged, We can add a `#Contribution` section to the bottom of this doc that outlines how to contribute code or non-code artifacts.

File: README.md [hunk 4]
```diff
@@ -1,235 +1,436 @@
...
+Your feedback, contributions, and issue reports will be integral to evolving this codebase from its Alpha phase into a mature and reliable node in the Kaspa network.
+
+## Installation
+  <details>
+  <summary>Building on Linux</summary>
+  
+  1. Install  general prerequisites
+
+      ```bash
+      sudo apt install build-essential \
```
Comment (coderofstuff) [inline README.md:21]: Please have this be a single line. In desktop, it looks like too much whitespace on the right when it's split.

File: README.md [hunk 5]
```diff
@@ -1,235 +1,436 @@
...
+      libssl-dev pkg-config 
+      ```
+
+  2. Install Protobuf (required for gRPC)
+  
+      ```bash
+      sudo apt install protobuf-compiler \
```
Comment (coderofstuff) [inline README.md:28]: Please have this be a single line. In desktop, it looks like too much whitespace on the right when it's split.

**Review by coderofstuff — CHANGES_REQUESTED** [review]
Thank you for making this effort to cleanup this README.md

I have some minor requests before I can approve this.

Note: I've verified that the content here is a superset (but is mostly the same) as the old wiki.

Notable additions are the intro and the extra links to some terms (borsh, for example)

File: kaspa.png
Comment (starkbamse) [reply kaspa.png:1]: Image is removed

File: README.md (hunk 4 above)
Comment (starkbamse) [reply README.md:21]: +1

File: README.md (hunk 5 above)
Comment (starkbamse) [reply README.md:28]: +1

File: README.md (hunk 1 above)
Comment (starkbamse) [reply README.md:4]: +1

File: kaspa.png
Comment (coderofstuff) [reply kaspa.png:1]: I mean, please remove it from the actual PR. It's still included in the `Files Changed` currently.

Comment (starkbamse) [reply kaspa.png:1]: Sorry, I totally thought I had already removed it from the PR. Removed now.

**Review by coderofstuff — APPROVED** [review]
LGTM

**Review by aspect — APPROVED** [review]
LGTM;  @coderofstuff many thanks for reviewing.

**Pull request merged**